src/time_util.rs

```diff
@@ -26,12 +28,37 @@ fn datetime_from_timestamp(context: &Timestamp) -> Option<DateTime<FixedOffset>>
 pub fn format_absolute_timestamp(timestamp: &Timestamp) -> String {
     static FORMAT_ITEMS: Lazy<Vec<chrono::format::Item>> =
         Lazy::new(|| StrftimeItems::new("%Y-%m-%d %H:%M:%S.%3f %:z").collect());
+    format_absolute_timestamp_with(timestamp, FORMAT_ITEMS.iter())
+}
+
+pub fn format_absolute_timestamp_with<'a, I, B>(timestamp: &Timestamp, items: I) -> String
+where
+    I: Iterator<Item = B> + Clone,
+    B: Borrow<chrono::format::Item<'a>>,
+{
     match datetime_from_timestamp(timestamp) {
-        Some(datetime) => datetime.format_with_items(FORMAT_ITEMS.iter()).to_string(),
+        Some(datetime) => datetime.format_with_items(items).to_string(),
         None => "<out-of-range date>".to_string(),
     }
 }
 
+/// Parses strftime-like format string into owned format items.
+pub fn parse_format_owned(format: &str) -> Option<Vec<chrono::format::Item<'static>>> {
```

Comment:
Agreed. Added newtype and split parsing function into `::parse()` and `.into_owned()`.

> I think it would also be less confusing to pass &FORMAT_ITEMS instead of FORMAT_ITEMS.iter(). Conceptually, at least to me, it's a single thing.

The underlying function requires `Iterator`, not `IntoIterator`. That's why. Anyway, our wrapper function now takes a newtype wrapper, so `.iter()` vs `&` is an implementation detail.